src/server/superPhramOrders.ts

```diff
@@ -0,0 +1,170 @@
+import * as fetch from "node-fetch";
+import { DataControl } from "@remult/angular";
+import { DateOnlyField, Entity, Field, IdEntity, ProgressListener, Remult} from "remult";
+import { Roles } from "../app/users/roles";
+import { LogisticsLog } from "../app/invoices-sent-to-logistics/logistics-log";
+import { DateOnlyValueConverter } from "remult/valueConverters";
+import { sendDataToFtp } from "../app/invoices-sent-to-logistics/sendFtp";
+import { LogisticsController } from "../app/invoices-sent-to-logistics/logistics-controller";
+import { callRivhit } from "../app/create-invoice/invoice";
+
+export async function callSuperPharm(api: string) {
+  const { SUPERPHARMSHOPKEY, SUPERPHARMSHOPURL } = process.env;
+
+  let headers = new fetch.Headers();
+  headers = {
+    Accept: "application/json",
+    Authorization: SUPERPHARMSHOPKEY,
+  };
+
+  let r = await fetch.default(SUPERPHARMSHOPURL + "/api/orders", {
+      method: "GET",
+      headers,
+    })
+    .then((response) => { return response.json()})
+    .catch((error) => {
+      throw error?.client_message || error;
+    });
+  return r.orders;
+}
+
+@Entity<OrdersSentToRivhit>("OrdersSentToRivhit", {
+  allowApiRead: Roles.admin,
+  defaultOrderBy: (self) => [
+    self.invoiceDate.descending(),
+    self.transmitDate.descending(),
+  ],
+})
+export class OrdersSentToRivhit extends IdEntity {
+  @DataControl({ width: "40" })
+  @Field({ caption: "סוג" })
+  documentType: number = 0;
+
+  @DataControl({ width: "60" })
+  @Field({ caption: "#" })
+  documentNumber: number = 0;
+
+  @DataControl({ width: "100" })
+  @DateOnlyField({
+    caption: "תאריך",
+    displayValue: (_, d) => d.toLocaleDateString(),
+  })
+  invoiceDate!: Date;
+
+  @DataControl({ width: "250" })
+  @Field({ caption: "לקוח" })
+  customerName: string = "";
+
+  @DataControl({ width: "250" })
+  @Field({ caption: "לקוח" })
+  order_id: string = "";
+
+  @DataControl({ width: "100" })
+  @Field({ caption: "סכום" })
+  amount: number = 0;
+
+  @DataControl({ width: "200" })
+  @Field({
+    allowApiUpdate: false,
+    caption: "שודר",
+    displayValue: (_, d) => {
+      let offset = d.getTimezoneOffset();
+      return new Date(d.valueOf() + offset * 60 * 1000).toLocaleString();
+      //d.toLocaleDateString();
+    },
+  })
+  transmitDate: Date = new Date();
+
+  @Field({ caption: "סטטוס" })
+  status: string = "";
+}
+
+let running = false;
+export async function checkForNewOrdersOnSuperpharm( remult?: Remult, progress?: ProgressListener) {
+  if (running) throw "check for new orders already running";
+  running = true;
+  const log = await remult.repo(LogisticsLog).findFirst({ createIfNotFound: true });
```

Comment:
You're using the same logistics log used by the other sync mechanism - meaning you might break it